Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -124,6 +104,28 @@ public class RepositoryPackageContainer: BasePackageContainer, CustomStringConve
             currentToolsVersion: currentToolsVersion
         )
     }
+    
+    // Compute the map of known versions.
+    private func knownVersions() throws -> [Version: String] {
+        try self.knownVersionsCache.memoize() {
+            let knownVersionsWithDuplicates = Git.convertTagsToVersionMap(try repository.tags())
+
+            return knownVersionsWithDuplicates.mapValues({ tags -> String in
+                if tags.count == 2 {
+                    // FIXME: Warn if the two tags point to different git references.
+                    return tags.first(where: { !$0.hasPrefix("v") })!
+                }
+                assert(tags.count == 1, "Unexpected number of tags")
+                return tags[0]
+            })
+        }
+    }
+    
+    public override func reversedVersions() throws -> [Version] {
+        try self.reversedVersionsCache.memoize() {
+            [Version](try self.knownVersions().keys).sorted().reversed()
```

Comment:
Also existing logic but would it be possible for `sorted` to sort in reverse order so `reversed()` is not needed?

Reply:
yes, this also caught my eye but wanted to minimize changes to existing logic. will create a follow-on PR?